core/src/main/java/org/fao/geonet/api/records/attachments/FilesystemStore.java

```diff
@@ -148,7 +149,7 @@ public MetadataResource putResource(final ServiceContext context, final String m
                                         Boolean approved) throws Exception {
         int metadataId = canEdit(context, metadataUuid, approved);
         Path filePath = getPath(context, metadataId, visibility, filename);
-        Files.copy(is, filePath);
+        Files.copy(is, filePath, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
The problem seem happening in `getPath` method:

https://github.com/geonetwork/core-geonetwork/blob/608c1b821467c3ef70d983813539c2b9eb6923ab/core/src/main/java/org/fao/geonet/api/records/attachments/FilesystemStore.java#L168

Checking a change to use the `approved` value to throw the exception only if not using the approved version.